src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -254,6 +345,25 @@ def _build_key_usage(self, ext):
             decipher_only
         )
 
+    def _build_subject_alt_name(self, ext):
+        gns = self._backend._ffi.cast(
+            "GENERAL_NAMES *", self._backend._lib.X509V3_EXT_d2i(ext)
+        )
+        assert gns != self._backend._ffi.NULL
+        # GCing this should be safe. See line 964 of v3_utl.c
```

Comment:
lol

On Mon, Apr 20, 2015 at 10:21 PM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/hazmat/backends/openssl/x509.py
> <https://github.com/pyca/cryptography/pull/1855#discussion_r28745721>:
>
> > @@ -254,6 +345,25 @@ def _build_key_usage(self, ext):
> >              decipher_only
> >          )
> >
> > +    def _build_subject_alt_name(self, ext):
> > +        gns = self._backend._ffi.cast(
> > +            "GENERAL_NAMES *", self._backend._lib.X509V3_EXT_d2i(ext)
> > +        )
> > +        assert gns != self._backend._ffi.NULL
> > +        # GCing this should be safe. See line 964 of v3_utl.c
>
> I didn't actually want this one to be in the PR at all. That was a
> reference to myself. Gone now :)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1855/files#r28745721>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084